os/pm/pm_changestate.c

```diff
@@ -300,11 +300,9 @@ int pm_changestate(int domain_indx, enum pm_state_e newstate)
 		pm_changeall(domain_indx, newstate);
 		pdom->state = newstate;
 		
-		/* Adjusting the sleep duration, so that system can revert to NORMAL
-		*  state when next Wifi keep alive signal is required to be sent. 
-		*/
 		if (newstate == PM_SLEEP) {
-			pm_adjust_sleep_duration();
+			/* Set the required wakeup timer from the g_pmTimer_activeList */
+			ret = pm_set_wakeup_timer();
```

Comment:
The threshold value for the timer that will put the main core to sleep should be checked here.

If the set wake-up timer interval value is less than the time during the main core actually sleeps, it is an unnecessary sleep.

Reply:
Please check this comment.

we need threshold